mission_control/include/mission_control/behavior.h

```diff
@@ -32,370 +32,65 @@
+        setStatus(BT::NodeStatus::RUNNING);
         break;
-      case TimeStamp::ABS_TIME:
-        // Construct the appropriate absolute datetime
-        calc_ptime = boost::posix_time::ptime(cur_ptime.date(), tout.time.time_of_day());
-        // Make sure to handle if the time is less than the current time (occurs next day)
-        if (calc_ptime.time_of_day() < cur_ptime.time_of_day())
-          calc_ptime += boost::gregorian::days(1);
-        nsec = (calc_ptime - cur_ptime).total_seconds();
+      case BT::NodeStatus::RUNNING:
+        setStatus(behaviorRunningProcess());
         break;
-      case TimeStamp::ABS_DATETIME:
-        if (tout.time < cur_ptime)
-        {
-          ROS_ERROR("Timeout occurs in the past; aborting mission");
-          break;
-        }
-        nsec = (tout.time - cur_ptime).total_seconds();
+      case BT::NodeStatus::SUCCESS:
+        halt();
         break;
-    };
-    return nsec;
-  }
-
-  bool startBehavior()
-  {
-    std::string xml_tag = getXmlTag();
-    switch (getType())
-    {
-      case BEHAVIOR_TYPE_MSG:
+      case BT::NodeStatus::FAILURE:
+        setStatus(BT::NodeStatus::FAILURE);
         break;
-      case BEHAVIOR_TYPE_SRV:
+      default:
+        throw std::logic_error("Unexpected state in ::tick()");
         break;
     }
-    return true;
-  }
-
-  bool stopBehavior()
-  {
-    std::string xml_tag = getXmlTag();
-    switch (getType())
-    {
-      case BEHAVIOR_TYPE_MSG:
-        break;
-      case BEHAVIOR_TYPE_SRV:
-        break;
-    }
-    return true;
-  }
-
- protected:
-  bool parseTimeStamps(std::list<BehaviorXMLParam>::iterator it)
-  {
-    bool retval = true;
-    bool pnts1 = true;
-    bool pnts2 = true;
-    std::string xmlParamTag = it->getXMLTag();
-    if (xmlParamTag.compare("when") == 0)
-    {
-      std::map<std::string, std::string> attribMap = it->getXMLTagAttribute();
-      if (attribMap.size() > 0)
-      {
-        std::map<std::string, std::string>::iterator attrib_it;
-        attrib_it = attribMap.begin();
-        pnts1 = parseNodeTimeStamp(attrib_it->second, it->getXMLTagValue(), m_when);
-      }
-      else
-      {
-        std::cout << "No attributes for when tag" << std::endl;
-      }
-    }
-    else if (xmlParamTag.compare("timeout") == 0)
-    {
-      std::map<std::string, std::string> attribMap = it->getXMLTagAttribute();
-      if (attribMap.size() > 0)
-      {
-        std::map<std::string, std::string>::iterator attrib_it;
-        attrib_it = attribMap.begin();
-        pnts2 = parseNodeTimeStamp(attrib_it->second, it->getXMLTagValue(), m_timeout);
-        m_timeout_ena = true;
-      }
-      else
-      {
-        std::cout << "No attributes for timeout tag" << std::endl;
-      }
-    }
-
-    if ((pnts1 == false) || (pnts2 == false)) retval = false;
-
-    return retval;
+    return status();
   }
-
-  bool parseNodeTimeStamp(std::string unit, std::string val, TimeStamp& t)
+  virtual void halt() override
```

Comment:
@patapufate hmm, looking at [`BT::AsyncActionNode`](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/6983ace9f722d31776ab4c5de857858cba5c052c/src/action_node.cpp#L199) implementation, I have the impression we shouldn't be overriding this method.

Reply:
Yes. You're right. I had overrided it taking into account the past architecture that the action will perform an "Abort" using halt() process. I removed it.